### PR TITLE
SUSE update chronyd max poll to support chrony.d

### PR DESCRIPTION
**Pull request by brett060102**
For chrony ntp servers may be listed in either chrony.conf or
in conf files in the chrony.d directory.
This is correctly handled in the oval, but the ansible
and bash remediation will only handles chrony.conf

To fix, find all chrony conf file and loop over those.

#### Description:

- extend ansible and bash remediation to handle conf files under chrony.d/ 
- on sle systems pidof is /bin/pidof so, we work the logic around that.

#### Rationale:

- chrony allow for servers to be specified in files under ./chrony.d. This is supported in the oval test code, but was not being handled by the remediation

### CODE REVIEW

**Comment by brett060102** [comment]
I am seeing two failed tests.
ci/prow/e2e-aws-rhcos4-high and ci/prow/e2e-aws-rhcos4-moderate  that look like it could be related to the
linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/ansible/shared.yml change. Amsible is exiting with but I am not sure why.

I am not sure what that test is really testing. If testing ntp.conf remediation, then the changes would not have effected that, If testing chrony.conf remediation, then it would have. If it is chrony, then we need to know that the chrony.cong being remediated look like so we can debug it.

File: linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/ansible/shared.yml [hunk 1]
```diff
@@ -11,35 +11,50 @@
...
+  loop: "{{ update_chrony_files.stdout_lines|list|flatten|unique }}"
+  when: chrony_conf_exist_result.stat.exists
+
 - name: Set the maxpoll values in {{{ chrony_conf_path }}}
   replace:
-    path: {{{ chrony_conf_path }}}
+    path: "{{ item }}"
     regexp: '(^(?:server|pool|peer)\s+((?!maxpoll).)*)$'
     replace: '\1 maxpoll {{ var_time_service_set_maxpoll }}\n'
+  loop: "{{ update_chrony_files.stdout_lines|list|flatten|unique} }"
```
Comment (jan-cerny) [inline linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/ansible/shared.yml:59]: white space problem

**Review by jan-cerny — APPROVED** [review]

**Review by jan-cerny — CHANGES_REQUESTED** [review]
It breaks the Ansible remediations:

```
[jcerny@thinkpad scap-security-guide{pr/9257}]$ python3 tests/automatus.py rule --remediate-using ansible --libvirt qemu:///system ssgts_rhel9 chronyd_or_ntpd_set_maxpoll
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2022-07-29-1127/test_suite.log
WARNING - Script chrony_nothing_done.fail.sh is not applicable on given platform
WARNING - Script ntp.pass.sh is not applicable on given platform
WARNING - Script ntp_wrong_maxpoll.fail.sh is not applicable on given platform
WARNING - Script ntp_multiple_misconfigured.fail.sh is not applicable on given platform
INFO - xccdf_org.ssgproject.content_rule_chronyd_or_ntpd_set_maxpoll
INFO - Script chrony.pass.sh using profile xccdf_org.ssgproject.content_profile_stig OK
INFO - Script chrony_no_pool_nor_servers.pass.sh using profile xccdf_org.ssgproject.content_profile_stig OK
INFO - Script chrony_one_pool_configured.pass.sh using profile xccdf_org.ssgproject.content_profile_stig OK
INFO - Script chrony_one_pool_misconfigured.fail.sh using profile xccdf_org.ssgproject.content_profile_stig OK
ERROR - Ansible playbook remediation run has exited with return code 2 instead of expected 0
ERROR - The remediation failed for rule 'xccdf_org.ssgproject.content_rule_chronyd_or_ntpd_set_maxpoll'.
INFO - Script chrony_one_pool_missing_parameter.fail.sh using profile xccdf_org.ssgproject.content_profile_stig OK
ERROR - Ansible playbook remediation run has exited with return code 2 instead of expected 0
ERROR - The remediation failed for rule 'xccdf_org.ssgproject.content_rule_chronyd_or_ntpd_set_maxpoll'.
INFO - Script chrony_one_server_misconfigured.fail.sh using profile xccdf_org.ssgproject.content_profile_stig OK
ERROR - Ansible playbook remediation run has exited with return code 2 instead of expected 0
ERROR - The remediation failed for rule 'xccdf_org.ssgproject.content_rule_chronyd_or_ntpd_set_maxpoll'.
```

Can you take a look into it?

File: linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/ansible/shared.yml (hunk 1 above)
Comment (brett060102) [reply linux_os/guide/services/ntp/chronyd_or_ntpd_set_maxpoll/ansible/shared.yml:59]: Thank you. I was just not seeing that.

**Comment by brett060102** [comment]
@jan-cerny 
I am not sure what is going one.  from the logs I see:
INFO - Script chrony_one_server_misconfigured.fail.sh using profile xccdf_org.ssgproject.content_profile_stig OK
28
ERROR - Rule evaluation resulted in fail, instead of expected pass during final stage 
29
ERROR - The check after remediation failed for rule 'xccdf_org.ssgproject.content_rule_chronyd_or_ntpd_set_maxpoll'.
30
INFO - Script chrony_one_pool_missing_parameter.fail.sh using profile xccdf_org.ssgproject.content_profile_stig OK
31
ERROR - Rule evaluation resulted in fail, instead of expected pass during final stage 
32
ERROR - The check after remediation failed for rule 'xccdf_org.ssgproject.content_rule_chronyd_or_ntpd_set_maxpoll'.
33
INFO - Script chrony.pass.sh using profile xccdf_org.ssgproject.content_profile_stig OK
34
INFO - Script chrony_one_pool_misconfigured.fail.sh using profile xccdf_org.ssgproject.content_profile_stig OK
35
ERROR - Rule evaluation resulted in fail, instead of expected pass during final stage 
36
ERROR - The check after remediation failed for rule 'xccdf_org.ssgproject.content_rule_chronyd_or_ntpd_set_maxpoll'.

which I think means that retest failed after remediating for:
chrony_one_pool_misconfigured.fail.sh 
chrony_one_server_misconfigured.fail.sh

 have check both on my test machine and I get a pass after remediation.

Any ideas about what I am missing.
Thanks

**Comment by brett060102** [comment]
Ansible tests pass in the Automatus CS9 / Run Tests (pull_request) but fail in the Automatus CS8 / Run Tests (pull_request).
I am not sure what the difference is between those two runs.

**Comment by brett060102** [comment]
@jan-cerny I have addressed the problem with the ansible code. Ansible tests now pass in the Automatus CS9 / Run Tests (pull_request) but fail in the Automatus CS8 / Run Tests (pull_request). They also pass in my local testing. So, I am not sure what the issue is with the CS8 tests.

**Comment by jan-cerny** [comment]
I think that the reason is that the Ansible tasks have a condition to not execute the tasks in a container environment (`- ansible_virtualization_type not in ["docker", "lxc", "openvz", "podman", "container"]`), which causes that all of the Ansible tasks in the remediation are skipped, because the GitHub CI job uses a container backend for testing the rules.

When using a virtual machine backend, everything passes:

```
[jcerny@thinkpad scap-security-guide{pr/9257}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel8 --remediate-using ansible chronyd_or_ntpd_set_maxpoll
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2022-08-02-1328/test_suite.log
WARNING - Script chrony_nothing_done.fail.sh is not applicable on given platform
WARNING - Script ntp.pass.sh is not applicable on given platform
WARNING - Script ntp_wrong_maxpoll.fail.sh is not applicable on given platform
WARNING - Script ntp_multiple_misconfigured.fail.sh is not applicable on given platform
INFO - xccdf_org.ssgproject.content_rule_chronyd_or_ntpd_set_maxpoll
INFO - Script chrony.pass.sh using profile xccdf_org.ssgproject.content_profile_stig OK
INFO - Script chrony_no_pool_nor_servers.pass.sh using profile xccdf_org.ssgproject.content_profile_stig OK
INFO - Script chrony_one_pool_configured.pass.sh using profile xccdf_org.ssgproject.content_profile_stig OK
INFO - Script chrony_one_pool_misconfigured.fail.sh using profile xccdf_org.ssgproject.content_profile_stig OK
INFO - Script chrony_one_pool_missing_parameter.fail.sh using profile xccdf_org.ssgproject.content_profile_stig OK
INFO - Script chrony_one_server_misconfigured.fail.sh using profile xccdf_org.ssgproject.content_profile_stig OK
```

So I think the Ansible changes are fine now.

**Review by jan-cerny — APPROVED** [review]

**Comment by brett060102** [comment]
Thank you.

**Pull request merged**